src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -870,9 +934,12 @@ void GoogleCalendarSyncAdaptor::purgeDataForOldAccount(int oldId, SocialNetworkS
 void GoogleCalendarSyncAdaptor::beginSync(int accountId, const QString &accessToken)
 {
     SOCIALD_LOG_DEBUG("beginning Calendar sync for Google, account" << accountId);
-    bool needCleanSync = !wasLastSyncSuccessful(accountId);
+    bool needCleanSync = false;
+    bool lastSyncSuccessful = wasLastSyncSuccessful(accountId, &needCleanSync);
     if (needCleanSync) {
-        SOCIALD_LOG_INFO("last sync was not successful; performing clean sync");
+        SOCIALD_LOG_INFO("performing clean sync");
+    } else if (!lastSyncSuccessful) {
+        SOCIALD_LOG_INFO("last sync was not successful, attempting to recover without clean sync");
```

Comment:
It looks like the bool returned by wasLastSyncSuccessful() is not used anymore, except for debugging. Is it still useful to store this value? Should the code document what are recoverable vs unrecoverable circumstances?